server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -111,7 +111,7 @@ public void execute(SearchContext context) {
         SearchHit[] hits = new SearchHit[context.docIdsToLoadSize()];
 
         List<FetchSubPhaseProcessor> processors = getProcessors(context.shardTarget(), fetchContext);
-        NestedDocuments nestedDocuments = context.getNestedDocuments();
+        NestedDocuments nestedDocuments = context.getQueryShardContext().getNestedDocuments();
```

Comment:
This is the type of change that we could make as a follow-up to keep this one contained. If we made this separately, there would probably be other things that catch our eyes in DefaultSearchContext to address with it. Also, we are not after fixing the fetch phase, so what are the gains? Could we leave the method in SearchContext for now and retrieve the MappingLookup from the MapperService in this case, and file it as a followup?

Reply:
Along the same lines, I wonder: if we avoid moving the method to QueryShardContext, do we still need to migrate NestedDocuments to MappingLookup, and to consequently move the method it uses from DocumentMapper to MappingLookup? It is a good change I think, but I don't see if it's required. In fact, QueryShardContext only needs hasNested which requires no changes as it's already part of MappingLookup? Am I missing anything?

Reply:
Makes sense to me. I'll have another look. I remember it being obvious to me at the time but I think all kinds of silly things.

Reply:
I've removed this change and added it as a follow up listed in #64663.